### PR TITLE
website/docs: add new doc about extra steps for hardening authentik

### PR DESCRIPTION
**Pull request by tanberry**
Added extra verbiage, from Jen's doc: https://www.notion.so/authentiksecurity/Hardening-docs-684229e00cdd49bbaa5c4ffd10c59ee2?v=684229e00cdd49bbaa5c4ffd10c59ee2

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

### CODE REVIEW

File: website/docs/security/security-hardening.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+[Expressions](../property-mappings/expression.mdx) allow super-users and other highly privileged users to create custom logic within authentik to modify its behaviour. Editing/creating these expressions is, by default, limited to super-users and any related events are fully logged.
+
+However, for further hardening, it is possible to prevent any user (even super-users) from using expressions to create or edit any objects. To do so, configure your deployment to block API requests to these endpoints:
+
+-   `/api/v3/policies/expression*`
+-   `/api/v3/propertymappings*`
+-   `/api/v3/managed/blueprints*`
+
+With these restrictions in place, expressions can only be edited using [Blueprints on the file system](https://docs.goauthentik.io/developer-docs/blueprints/#storage---file). And of course, implement restricted access to the file system itself.
```
Comment (kensternberg-authentik) [inline website/docs/security/security-hardening.md:17]: Suggestion: "Take care to restrict access to the file system itself."  That "... and of course" seems a bit off-hand.

File: website/docs/security/security-hardening.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+### Blueprints
+
+Blueprints allow for templating and managing the authentik configuration as code. Just like expressions, they can only be created/edited by super-users or users with specific permissions assigned to them. However, because they interact with the authentik API on a lower level, they can create other objects.
+
+To prevent any user from creating/editing blueprints, block API requests to this endpoint:
+
+-   `/api/v3/managed/blueprints*`
+
+With these restrictions in place, expressions can only be edited using [Blueprints on the file system](https://docs.goauthentik.io/developer-docs/blueprints/#storage---file).
```
Comment (kensternberg-authentik) [inline website/docs/security/security-hardening.md:27]: Does this duplicate line 17 above?  Same API URL, same sentence, only minus the expressions and mappings bits.

**Review by kensternberg-authentik — APPROVED** [review]

File: website/docs/security/security-hardening.md (hunk 2 above)
Comment (tanberry) [reply website/docs/security/security-hardening.md:27]: Yeah I need to ask @BeryJu about this line, and the third that is similar.

Comment (BeryJu) [reply website/docs/security/security-hardening.md:27]:
```suggestion
With these restrictions in place, Blueprints can only be edited via [the file system](https://docs.goauthentik.io/developer-docs/blueprints/#storage---file).
```

**Pull request merged**